package/piksi_system_daemon/src/whitelists.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2017 Swift Navigation Inc.
+ * Contact: Gareth McMullin <gareth@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+/* https://swiftnav.hackpad.com/SBP-Message-Whitelist-Design-toM7WBRqq3C */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "settings.h"
+#include "whitelists.h"
+
+enum port {
+  PORT_UART0,
+  PORT_UART1,
+  PORT_GSERIAL,
+  PORT_ETHERNET,
+  PORT_MAX
+};
+
+const char *setting_names[PORT_MAX] = {
+  "uart0", "uart1", "usb", "ethernet",
+};
+
+/* Whitelist settings are kept as formatted strings of message ids and
+ * rate dividers.  Strings are parsed in whilelist_notify()
+ *
+ * Examples:
+ * ""
+ *  - All messages are sent
+ * "65535"
+ *  - Only the heartbeat message is sent (message ID 65535)
+ * "1234,5678/2,3456/10"
+ *  - Message 1234 is sent at full rate
+ *  - Message 5678 is sent at half rate
+ *  - Message 3456 is sent at 1/10 rate
+ */
+char wl[PORT_MAX][256] ={
+  [PORT_UART0] = "68,72,73,74,65535",
```

Comment:
Maybe we should remove the config files in `board/piksiv3/rootfs-overlay/etc` and let `piksi_system_daemon` create them. Otherwise I expect someone will modify them trying to change the defaults.